### ISSUE TITLE
[Bug]: react/no-unknown-property reports valid CSS Transition event handlers as unknown

### ISSUE THREAD

**Issue by rentalhost** [yannickcr/eslint-plugin-react#3993]
### Is there an existing issue for this?

- [x] I have searched the existing issues and my issue is unique
- [x] My issue appears in the command-line and not only in the text editor

### Description Overview

The `no-unknown-property` rule only recognizes `onTransitionEnd` and `onTransitionEndCapture` from the [CSS Transition events](https://developer.mozilla.org/en-US/docs/Web/API/Element#transition_events), but does not recognize the remaining standard transition event handlers.

These are valid DOM events listed on MDN's `Element` interface and are properly typed in `@types/react` ([DefinitelyTyped source](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0.1.450/types/react/index.d.ts)).

**Missing event handlers (6):**

- `onTransitionCancel`
- `onTransitionCancelCapture`
- `onTransitionRun`
- `onTransitionRunCapture`
- `onTransitionStart`
- `onTransitionStartCapture`

Reproduction:

```jsx
// All of these trigger "Unknown property" warnings:
<div onTransitionCancel={() => {}} />
<div onTransitionCancelCapture={() => {}} />
<div onTransitionRun={() => {}} />
<div onTransitionRunCapture={() => {}} />
<div onTransitionStart={() => {}} />
<div onTransitionStartCapture={() => {}} />
```

These should be added to the `DOM_PROPERTY_NAMES_TWO_WORDS` array in [`lib/rules/no-unknown-property.js`](https://github.com/jsx-eslint/eslint-plugin-react/blob/v7.37.5/lib/rules/no-unknown-property.js), alongside the already recognized `onTransitionEnd` and `onTransitionEndCapture`.

**Note:** there may be other missing event handlers following the same pattern (e.g. `onAnimationCancel` / `onAnimationCancelCapture`). The scope of this issue is focused on the transition events, but a broader audit of the event handler list against the current MDN `Element` interface and `@types/react` might be worthwhile.

### Expected Behavior

The rule should recognize all standard CSS Transition event handlers without reporting them as unknown properties. Using `onTransitionCancel`, `onTransitionRun`, or `onTransitionStart` (and their `Capture` variants) on a DOM element should not trigger a warning.

### eslint-plugin-react version

v7.37.5

### eslint version

v10.1.0

### node version

v24.14.0

**Comment by ternaus** [comment]
The six CSS transition event handlers from this report are included in [@ternaus/eslint-plugin-react v8.0.0](https://github.com/ternaus/eslint-plugin-react/releases/tag/v8.0.0): `onTransitionCancel`, `onTransitionRun`, and `onTransitionStart`, with their capture variants. The release is part of an independent ESLint 10 / React 19+ continuation; scope and migration details are in [#4036](https://github.com/jsx-eslint/eslint-plugin-react/issues/4036).
